frequenz_resampling/_rust_backend.pyi

```diff
@@ -0,0 +1,81 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+__all__ = "Resampler", "ResamplingFunction"
+
+from enum import Enum
+from datetime import datetime, timedelta
+from typing import Optional
+
+class ResamplingFunction(Enum):
+    """
+    The ResamplingFunction enum represents the different resampling functions
+    that can be used to resample a time series.
+    """
+
+    Average = 0
+    """Calculates the average of all samples in the time step (ignoring None values)"""
+    Sum = 1
+    """Calculates the sum of all samples in the time step (ignoring None values)"""
+    Max = 2
+    """Calculates the maximum of all samples in the time step"""
+    Min = 3
+    """Calculates the minimum of all samples in the time step"""
+    Last = 3
+    """Returns the last sample in the time step"""
+    Count = 3
+    """Counts the number of samples in the time step"""
```

Comment:
Is this correct? If not, you can avoid this in the future by using [`@unique`](https://docs.python.org/3/library/enum.html#enum.unique) (and maybe even [`auto()`](https://docs.python.org/3/library/enum.html#enum.auto)).

Reply:
I just realized that the interface to a Rust Enum is not a Python Enum but just a class with the options as class variables of itself: 

```python
ResamplingFunction.Sum.Sum.Max
>>> ResamplingFunction.Max
```

Reply:
I just realized, the value returned by pyo3 is not an enum type. I will come up with another type description.

Reply:
I changed it to a normal class and added the options as class variables without values.

Reply:
OK, in the future if there is a way to make pyo3 use a `Enum` that would be great.

Reply:
* https://github.com/frequenz-floss/frequenz-resampling-rs/issues/4